cert.go

```diff
@@ -208,6 +208,79 @@ func (m *mkcert) newCA() {
 	log.Printf("Created a new local CA at \"%s\" 💥\n", m.CAROOT)
 }
 
+func (m *mkcert) signCSR() {
+	csrPEMBlock, err := ioutil.ReadFile(m.csr)
+	fatalIfErr(err, "failed to read the CSR")
+	csrDERBlock, _ := pem.Decode(csrPEMBlock)
+	if csrDERBlock == nil {
+		log.Fatalln("ERROR: failed to read the CSR: unexpected content")
+	}
+	if csrDERBlock.Type != "CERTIFICATE REQUEST" {
+		log.Fatalln("ERROR: failed to read the CSR: Must be CERTIFICATE REQUEST not " + csrDERBlock.Type)
+	}
+	csr, err := x509.ParseCertificateRequest(csrDERBlock.Bytes)
+	fatalIfErr(err, "failed to parse the CA key")
+
+	err = csr.CheckSignature()
+	fatalIfErr(err, "invalid CSR signature")
+
+	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
+	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+	fatalIfErr(err, "failed to generate serial number")
+
+	tpl := &x509.Certificate{
+		SerialNumber:       serialNumber,
+		Subject:            csr.Subject,
+		PublicKeyAlgorithm: csr.PublicKeyAlgorithm,
+		Version:            csr.Version,
+		Extensions:         csr.Extensions,
+		ExtraExtensions:    csr.ExtraExtensions,
```

Comment:
This is not going to work: `ExtraExtensions` is for marshaling and `Extensions` is for parsing, so `csr.ExtraExtensions` will be empty and `tpl.Extensions` will be ignored.

But if you set `tpl.ExtraExtensions` to `csr.Extensions`, there's no need to copy all the SAN values below because all requested extensions will be copied.

Mailed https://go-review.googlesource.com/c/go/+/160898 to improve the docs.